### PR TITLE
Bluetooth: Mesh: Allow sending SNB and PRB in parallel 

### PR DESCRIPTION
**Pull request by PavelVPV**
Private Beacons and Secure Network Beacons are not mutually exclusive by
spec. This means both beacons can be sent simultaneously. For both
beacons spec defines their own observation intervals and related
parameters, which means we need to monitor both beacons types
separately.

Additionally this PR fixes 2 more issues:
- restoring Private Beacon state after reboot
- disabling processing of received Private Beacons when mesh-1.1 is disabled

### CODE REVIEW

File: subsys/bluetooth/mesh/beacon.c [hunk 1]
```diff
@@ -104,11 +138,13 @@ static void secure_beacon_create(struct bt_mesh_subnet *sub,
...
 
-	net_buf_simple_add_mem(buf, sub->auth, 8);
+	net_buf_simple_add_mem(buf, sub->secure_beacon.auth, 8);
 
 	LOG_DBG("net_idx 0x%04x flags 0x%02x NetID %s", sub->net_idx, flags,
 		bt_hex(keys->net_id, 8));
-	LOG_DBG("IV Index 0x%08x Auth %s", bt_mesh.iv_index, bt_hex(sub->auth, 8));
+	LOG_DBG("IV Index 0x%08x Auth %s", bt_mesh.iv_index, bt_hex(sub->secure_beacon.auth, 8));
+
+	return 0;
```
Comment (alxelax) [inline subsys/bluetooth/mesh/beacon.c:147]: is it really required?

File: subsys/bluetooth/mesh/subnet.h [hunk 2]
```diff
@@ -71,7 +75,9 @@ struct bt_mesh_subnet {
 		uint8_t identity[16];  /* IdentityKey */
 	#endif
 		uint8_t beacon[16];      /* BeaconKey */
+#if defined(CONFIG_BT_MESH_V1d1)
```
Comment (alxelax) [inline subsys/bluetooth/mesh/subnet.h:78]: Should it be `CONFIG_BT_MESH_PRIV_BEACONS` instead?

**Review by alxelax — COMMENTED** [review]
LGTM, just minor questions

File: subsys/bluetooth/mesh/beacon.c (hunk 1 above)
Comment (PavelVPV) [reply subsys/bluetooth/mesh/beacon.c:147]: Yes, to align signature of `secure_beacon_create` and `private_beacon_create`

File: subsys/bluetooth/mesh/subnet.h (hunk 2 above)
Comment (PavelVPV) [reply subsys/bluetooth/mesh/subnet.h:78]: Mesh-1.1 node shall be able to process private beacons. `CONFIG_BT_MESH_PRIV_BEACONS` only enables transmission of the private beacons.

**Review by alxelax — APPROVED** [review]

**Review by Andrewpini — APPROVED** [review]
LGTM👍

**Pull request merged**